src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/MLDsaAsymmetricAlgorithmWrapper.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace System.Security.Cryptography.Cose
+{
+    [Experimental(Experimentals.PostQuantumCryptographyDiagId)]
+    internal sealed class MLDsaAsymmetricAlgorithmWrapper : AsymmetricAlgorithm
+    {
+        internal MLDsa WrappedKey { get; }
+
+        public MLDsaAsymmetricAlgorithmWrapper(MLDsa wrappedKey)
+        {
+            WrappedKey = wrappedKey;
+        }
+    }
```

Comment:
Since this is internal, we can avoid making more types by just having a different property for each key type.  e.g.

```c#
internal sealed class AsymmetricAlgorithmWrapper : AsymmetricAlgorithm
{
    internal MLDsa MLDsa { get; }
    internal SlhDsa SlhDsa { get; }
    internal CompositeMLDsa ComositeMLDsa { get; }

    internal AsymmetricAlgorithmWrapper(MLDsa key);
    internal AsymmetricAlgorithmWrapper(SlhDsa key);
    internal AsymmetricAlgorithmWrapper(CompositeMLDsa key);
}
```